ConsoleApp5/UserDialogueHandler.cs

```diff
@@ -11,16 +11,21 @@ namespace ConsoleApp5
 {
     public class UserDialogueHandler
     {
-        public static int ReadInt(string message)
+        public static long ReadInt(string message, long minValue, long maxValue)
         {
-            var isCorrect = true;
-            int userInputInt;
+            long userInputLong;
             do
             {
-                Console.WriteLine(message);
-                isCorrect = int.TryParse(Console.ReadLine(), out userInputInt);
-            } while (!isCorrect || userInputInt > 3 || userInputInt < 1);
-            return userInputInt;
+                Console.Write(message);
+                long.TryParse(Console.ReadLine(), out userInputLong);
```

Comment:
nie uzywasz w ogole wartosci zwracanej z tego TryParse. Jak parsowanie sie nie uda to `userInputLong` bedzie defaultem longa czyli 0.

Ta metoda ma czytac input a ty jeszcze dodatkowo dodales walidacje samej wartosci, co lamie srp